ffi/native/dart_vlc.cpp

```diff
@@ -25,374 +28,342 @@ extern "C" {
-    std::vector<std::string> args;
-    for (int index = 0; index < commandLineArgumentsCount; index++) args.emplace_back(commandLineArguments[index]);
-    Player* player = players->get(id, args);
-    player->videoWidth = videoWidth;
-    player->videoHeight = videoHeight;
-    player->onPlay([=]() -> void {
-        Player_onPlayPauseStop(player->state);
-    });
-    player->onPause([=]() -> void {
-        Player_onPlayPauseStop(player->state);
-    });
-    player->onStop([=]() -> void {
-        Player_onPlayPauseStop(player->state);
-        Player_onPosition(player->state);
-    });
-    player->onComplete([=]() -> void {
-        Player_onComplete(player->state);
-    });
-    player->onVolume([=](float _) -> void {
-        Player_onVolume(player->state);
-    });
-    player->onRate([=](float _) -> void {
-        Player_onRate(player->state);
-    });
-    player->onPosition([=](int _) -> void {
-        Player_onPosition(player->state);
-    });
-    player->onOpen([=](VLC::Media _) -> void {
-        Player_onOpen(player->state);
-    });
-    player->onPlaylist([=]() -> void {
-        Player_onOpen(player->state);
+static char** g_metas_ptr = nullptr;
+static size_t g_metas_size = 0;
+static char** g_devices_ptr = nullptr;
+static size_t g_devices_size = 0;
+static char** g_equalizer_ptr = nullptr;
+static size_t g_equalizer_size = 0;
+
+DLLEXPORT void PlayerCreate(int32_t id, int32_t video_width,
+                            int32_t video_height,
+                            int32_t commandLineArgumentsCount,
+                            const char** commandLineArguments) {
+  std::vector<std::string> args{};
+  for (int32_t index = 0; index < commandLineArgumentsCount; index++)
+    args.emplace_back(commandLineArguments[index]);
+  Player* player = g_players->Get(id, args);
+  player->SetVideoWidth(video_width);
+  player->SetVideoHeight(video_height);
+  player->OnPlay([=]() -> void { OnPlayPauseStop(id, player->state()); });
+  player->OnPause([=]() -> void { OnPlayPauseStop(id, player->state()); });
+  player->OnStop([=]() -> void {
+    OnPlayPauseStop(id, player->state());
+    OnPosition(id, player->state());
+  });
+  player->OnComplete([=]() -> void { OnComplete(id, player->state()); });
+  player->OnVolume([=](float) -> void { OnVolume(id, player->state()); });
+  player->OnRate([=](float) -> void { OnRate(id, player->state()); });
+  player->OnPosition([=](int32_t) -> void { OnPosition(id, player->state()); });
+  player->OnOpen([=](VLC::Media) -> void { OnOpen(id, player->state()); });
+  player->OnPlaylist([=]() -> void { OnOpen(id, player->state()); });
+#ifdef _WIN32
+/* Windows: Texture & flutter::TextureRegistrar */
+#else
+  /* Linux: decodeImageFromPixels & NativePorts */
+  if (player->video_width() > 0 && player->video_height() > 0) {
+    player->OnVideo([=](uint8_t* frame) -> void {
+      OnVideo(player->video_width() * player->video_height() * 4,
+              player->state(), frame);
```

Comment:
missing `id` argument

Reply:
@namniav I couldn't notice it because I was building on Windows. Fixed on `master`.